### ISSUE TITLE
Can't build OWLTools

### ISSUE THREAD

**Issue by rlwalls2008** [owlcollab/owltools#115]
When I tried to build OWLTools using Maven, after about 6 minutes of doing stuff I get a whole bunch of errors and a build fail message. I have attached a text file with the list of errors, in case that matters.

**Comment by rlwalls2008** [comment]
Actually, I can't attach a file, so I will just paste the beginning of the list here:

100 errors
100 warnings
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] OWLTools-Parent .................................... SUCCESS [  1.103 s]
[INFO] OWLTools-Core ...................................... FAILURE [05:05 min]
[INFO] Lego ............................................... SKIPPED
[INFO] OWLTools-Annotation ................................ SKIPPED
[INFO] OWLTools-Oort ...................................... SKIPPED
[INFO] OWLTools-Sim ....................................... SKIPPED
[INFO] OWLTools-Web ....................................... SKIPPED
[INFO] OWLTools-Solr ...................................... SKIPPED
[INFO] OWLTools-Runner .................................... SKIPPED
[INFO] OWLTools-NCBI ...................................... SKIPPED
[INFO] Golr-Client ........................................ SKIPPED
[INFO] MolecularModelServer ............................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 05:06 min
[INFO] Finished at: 2015-04-04T15:11:06-07:00
[INFO] Final Memory: 35M/532M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-javadoc-plugin:2.10.1:jar (attach-javadocs) on project OWLTools-Core: MavenReportException: Error while creating archive:
[ERROR] Exit code: 1 - /Users/rwalls-iplant/owltools/OWLTools-Core/src/main/java/owltools/cli/Opts.java:70: warning: no description for @param
[ERROR] * @param opt
[ERROR] ^
[ERROR] /Users/rwalls-iplant/owltools/OWLTools-Core/src/main/java/owltools/cli/Opts.java:169: warning: no description for @param
[ERROR] * @param params
[ERROR] ^
[ERROR] /Users/rwalls-iplant/owltools/OWLTools-Core/src/main/java/owltools/cli/Opts.java:170: warning: no description for @param
[ERROR] * @param desc
[ERROR] ^
[ERROR] /Users/rwalls-iplant/owltools/OWLTools-Core/src/main/java/owltools/diff/ReasonerDiff.java:27: warning: no description for @param
[ERROR] * @param newAxioms
[ERROR] ^
[ERROR] /Users/rwalls-iplant/owltools/OWLTools-Core/src/main/java/owltools/diff/ReasonerDiff.java:28: warning: no description for @param
[ERROR] * @param removedInferredAxioms
[ERROR] ^
...

**Comment by hdietze** [comment]
Thanks for reporting the problem.
What java and maven version are you using? 
These seem to be *non*-fatal Javadoc warnings. That should not cause the build to stop.

On a related note: we have a continuous integration test running for owltools. At the moment it compiles and passes all tests during the maven build. You can see the current status here:
http://build.berkeleybop.org/job/owltools/

**Comment by rlwalls2008** [comment]
I am using Maven 3.3.1 and Java 1.8.0_20. I had to set JAVA_HOME to point to the latest version before I could build.

It definitely failed to build! I can't use any of the tools.

**Comment by hdietze** [comment]
I can reproduce the problem using Java8. There are two possible work-arounds until we fix the underlying issue:
1. Use the pre-built owltools/oort from [Jenkins](http://build.berkeleybop.org/userContent/owltools/)
2. Skip the javadoc during the owltools build: `-Dmaven.javadoc.skip=true`

**Comment by hdietze** [comment]
The proper solution is to fix the Javadoc errors and also have an integration test using Java 8, in addition to the current Java 7 one in Jenkins.

**Comment by rlwalls2008** [comment]
Thanks! I will try to use the pre-built version. I have no desire to build it myself and just want to use a standard build. It might help to have a link to that on the home page here, if that will continue to be a viable option. 

I do realize that this github site is still under construction.

**Comment by hdietze** [comment]
The installation details are available in the OWLTools wiki. Unfortunately that is not very prominent on the landing page: [Installing OWLTools](https://github.com/owlcollab/owltools/wiki/Install-OWLTools)

**Comment by rlwalls2008** [comment]
Ah...  You just made my life a lot easier!

**Comment by fbastian** [comment]
Isn't it simply this issue? http://stackoverflow.com/questions/15886209/maven-is-not-working-in-java-8-when-javadoc-tags-are-incomplete

The fix is a one-liner in pom.xml.

**Comment by hdietze** [comment]
Yes that seems to be the issue. It is necessary to use a Maven profile, otherwise the Java 7 built fails.
But we still should run the integration tests both on Java 7 and Java 8.
